Some new failing tests

**Linux:**
```
1/3 Test #423: test/grdgdal/gdal_nn.sh ..........***Failed    0.51 sec
Set GMT_SESSION_NAME = 22582
malloc(): memory corruption
gmtest: line 57: 22635 Aborted                 (core dumped) "$@"
ERROR: gmtest:61
memtrack errors: 0
exit status: 1

2/3 Test #517: test/greenspline/gspline_4.sh ....***Failed    0.72 sec
Set GMT_SESSION_NAME = 22586
grdcontour [WARNING]: No contours within specified -L range
gawk: fatal: cannot open file `dump' for reading (No such file or directory)
ERROR: gspline_4.sh:27
memtrack errors: 0
exit status: 1
```

**macOS**
```
1/3 Test #229: test/api/apicubeplot.sh ...........***Failed    0.32 sec
Set GMT_SESSION_NAME = 48711
gmtest: line 57: 48762 Abort trap: 6           "$@"
ERROR: gmtest:57
memtrack errors: 0
exit status: 1

2/3 Test #454: test/grdinterpolate/onelayer.sh ...***Failed    0.39 sec
Set GMT_SESSION_NAME = 48713
grdinterpolate [ERROR]: Grid x increment <= 0.0
grdinterpolate [ERROR]: Grid y increment <= 0.0
grdinterpolate (gmtapi_init_grdheader): Please select compatible -R and -I values
grdcontour [WARNING]: Grid has wrong registration type. Switching from gridline to pixel registration
Assertion failed: ((wesn[XLO]+GMT_CONV4_LIMIT*header->inc[GMT_X]) >= header->wesn[XLO] && (wesn[XHI]-GMT_CONV4_LIMIT*header->inc[GMT_X]) <= header->wesn[XHI]), function gmtnc_grd_prep_io, file ../src/gmt_nc.c, line 1373.
gmtest: line 57: 48763 Abort trap: 6           "$@"
ERROR: gmtest:57
memtrack errors: 0
exit status: 1
```

**Windows**:
```

4/9 Test #517: test/greenspline/gspline_4.sh .....***Failed    7.89 sec
Set GMT_SESSION_NAME = 42283
grdcontour [WARNING]: No contours within specified -L range
gawk: fatal: cannot open file `dump' for reading (No such file or directory)
ERROR: gspline_4.sh:27
memtrack errors: 0
exit status: 1
```

BTW, I also get some warnings during compilation:
```
../../src/gmt_api.c:5326:20: warning: unused variable 'new' [-Wunused-variable]
        bool done = true, new = false;
                          ^
../../src/gmt_api.c:5327:27: warning: unused variable 'layer' [-Wunused-variable]
        uint64_t row, col, kol, layer, row_out, i0, i1, j0, j1, k0, k1, ij, ij_orig;
                                ^
../../src/gmt_api.c:5329:9: warning: unused variable 'size' [-Wunused-variable]
        size_t size;
               ^
../../src/gmt_api.c:5834:20: warning: unused variable 'row_by_row' [-Wunused-variable]
        bool done = true, row_by_row;
                          ^
../../src/gmt_api.c:6011:63: warning: variable 'here' is uninitialized when used here [-Wuninitialized]
                                        api_put_val (&(M_obj->data), ij, (double)U_obj->data[ijp+here]);
                                                                                                 ^~~~
../../src/gmt_api.c:5837:33: note: initialize the variable 'here' to silence this warning
        uint64_t k, n_layers_used, here, save_n_bands;
                                       ^
                                        = 0
5 warnings generated.
[90/263] Building C object src/CMakeFiles/gmtlib.dir/gmt_grdio.c.o
../../src/gmt_grdio.c:2542:92: warning: unused parameter 'start' [-Wunused-parameter]
GMT_LOCAL void gmtgrdio_pad_cube_off_sub (struct GMT_GRID *G, gmt_grdfloat *data, uint64_t start) {
                                                                                           ^
../../src/gmt_grdio.c:2542:16: warning: unused function 'gmtgrdio_pad_cube_off_sub' [-Wunused-function]
GMT_LOCAL void gmtgrdio_pad_cube_off_sub (struct GMT_GRID *G, gmt_grdfloat *data, uint64_t start) {
               ^
2 warnings generated.
[111/263] Building C object src/CMakeFiles/gmtlib.dir/gmt_support.c.o
../../src/gmt_support.c:12644:77: warning: unused parameter 'direction' [-Wunused-parameter]
int gmt_cube_BC_set (struct GMT_CTRL *GMT, struct GMT_CUBE *U, unsigned int direction) {
                                                                            ^
1 warning generated.
[171/263] Building C object src/CMakeFiles/gmtlib.dir/greenspline.c.o
../../src/greenspline.c:1441:8: warning: unused variable 'this_env' [-Wunused-variable]
        char *this_env = NULL;
              ^
1 warning generated.
```

Here are debugging messages of gspline_4.sh:
```
1/1 Test #517: test/greenspline/gspline_4.sh ....***Failed    0.64 sec
Set GMT_SESSION_NAME = 9727
gmtset (gmtlib_free_tmp_arrays): tried to free unallocated memory
gmt [DEBUG]: Exit:  gmt_reload_settings
gmt [DEBUG]: Enter: gmtlib_plot_C_format
gmt [DEBUG]: Exit:  gmtlib_plot_C_format
gmt [DEBUG]: Enter: gmtinit_get_history
gmt [DEBUG]: Initialize FFTW with 2 threads.
gmt [DEBUG]: GMT_Create_Session initialized GMT structure
gmt [DEBUG]: Loading core GMT shared library: libgmt.so
gmt [DEBUG]: Shared Library # 0 (core). Path = libgmt.so
gmt [DEBUG]: Loading GMT plugins from: /home/runner/work/gmt/gmt/build/src/plugins
gmt [DEBUG]: Shared Library # 1 (supplements). Path = /home/runner/work/gmt/gmt/build/src/plugins/supplements.so
gmt [DEBUG]: Local file /home/runner/.gmt/server/gmt_data_server.txt found
gmt [DEBUG]: File /home/runner/.gmt/server/gmt_data_server.txt less than 24 hours old, refresh is premature.
gmt [DEBUG]: Load contents from /home/runner/.gmt/server/gmt_data_server.txt
gmt [DEBUG]: Local file /home/runner/.gmt/server/gmt_hash_server.txt found
gmt [DEBUG]: File /home/runner/.gmt/server/gmt_hash_server.txt less than 24 hours old, refresh is premature.
gmt [DEBUG]: Revised options: -R12/32/0/6/5/10 -I0.25 -G3D_%05.2lf.grd @Table_5_23.txt -Sr0.85 -D5
gmt (gmtlib_free_tmp_arrays): tried to free unallocated memory
greenspline [DEBUG]: History: Process -R12/32/0/6/5/10
greenspline [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt
greenspline [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/work/gmt/gmt/test/greenspline
greenspline [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt/cache
greenspline [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt/server
greenspline [DEBUG]: Remote file @Table_5_23.txt exists locally as /home/runner/.gmt/cache/Table_5_23.txt
greenspline [DEBUG]: Replace file @Table_5_23.txt with /home/runner/.gmt/cache/Table_5_23.txt
greenspline [INFORMATION]: Enable all available threads (up to 2)
greenspline [DEBUG]: gmtapi_init_import: Passed family = Data Table and geometry = Point
greenspline [DEBUG]: Given full path to file /home/runner/.gmt/cache/Table_5_23.txt
greenspline [DEBUG]: Found readable file /home/runner/.gmt/cache/Table_5_23.txt
greenspline [DEBUG]: Object ID 0 : Registered Data Table File /home/runner/.gmt/cache/Table_5_23.txt as an Input resource with geometry Point [n_objects = 1]
greenspline (gmtapi_init_import): tried to free unallocated memory
greenspline [DEBUG]: gmtapi_init_import: Added 1 new sources
greenspline [DEBUG]: GMT_Init_IO: Returned first Input object ID = 0
greenspline [DEBUG]: GMT_Begin_IO: Mode value 1 not considered (ignored)
greenspline [DEBUG]: GMT_Begin_IO: Initialize record-by-record access for Input
greenspline [DEBUG]: gmtapi_next_io_source: Selected object 0
greenspline [DEBUG]: Given full path to file /home/runner/.gmt/cache/Table_5_23.txt
greenspline [DEBUG]: Found readable file /home/runner/.gmt/cache/Table_5_23.txt
greenspline [DEBUG]: Found readable file /home/runner/.gmt/cache/Table_5_23.txt
greenspline [INFORMATION]: Reading Data Table from file /home/runner/.gmt/cache/Table_5_23.txt
greenspline [DEBUG]: GMT_Begin_IO: Input resource access is now enabled [record-by-record]
greenspline [INFORMATION]: Read input data and check for data constraint duplicates
greenspline [DEBUG]: Source col types: (Number,Number,Number,Number)
greenspline [DEBUG]: ASCII source scanned: Numerical columns: 4, Trailing text: N, Record type: Numerical only
greenspline [DEBUG]: GMT_End_IO: Input resource access is now disabled
greenspline [INFORMATION]: Found 36 unique data constraints
greenspline [INFORMATION]: Distance between the closest constraints:  1.11803398875]
greenspline [INFORMATION]: Distance between most distant constraints: 38.8458491991]
greenspline [INFORMATION]: Found 36 data points, yielding a 36 by 36 set of linear equations
greenspline [DEBUG]: Grid/Image dimensions imply w/e/s/n = 12/32/0/6, inc = 0.25/0.25, gridline registration, n_layers = 1
greenspline [INFORMATION]: Cartesian input grid
greenspline [DEBUG]: Chosen boundary condition for all edges: natural
greenspline [INFORMATION]: Cartesian input grid
greenspline [DEBUG]: Object ID 1 : Registered Cube Memory Reference 5644b98de250 as an Input resource with geometry Not Set [n_objects = 2]
greenspline [DEBUG]: Successfully created a new Cube container
greenspline [DEBUG]: p_val = 2.38048
greenspline [INFORMATION]: Normalization mode: Remove mean and normalize data
greenspline [INFORMATION]: 2-D Normalization coefficients: zoff = 6.66111 xslope = 0 xmean = 0 yslope = 0 ymean = 0 range = 16.5389
greenspline [INFORMATION]: Square matrix A (size 36 x 36) requires 10.1 kb
greenspline [INFORMATION]: Build square linear system Ax = b using Regularized Cartesian spline in tension [3-D]
greenspline [INFORMATION]: Solve linear equations by Gauss-Jordan elimination
greenspline [INFORMATION]: Evaluate spline at 42525 equidistant output locations
greenspline [DEBUG]: Object ID 2 : Registered Cube File 3D_00.00.grd as an Output resource with geometry Not Set [n_objects = 3]
greenspline [DEBUG]: gmtapi_begin_io: Output resource access is now enabled [container]
greenspline [DEBUG]: gmtapi_export_cube: Passed ID = 2 and mode = 0
greenspline [INFORMATION]: Writing cube to file 3D_%05.2lf.grd
greenspline [DEBUG]: Object ID 3 : Registered Grid File 3D_05.00.grd as an Output resource with geometry Surface [n_objects = 4]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 3 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_05.00.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 4 : Registered Grid File 3D_05.25.grd as an Output resource with geometry Surface [n_objects = 5]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 4 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_05.25.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 5 : Registered Grid File 3D_05.50.grd as an Output resource with geometry Surface [n_objects = 6]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 5 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_05.50.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 6 : Registered Grid File 3D_05.75.grd as an Output resource with geometry Surface [n_objects = 7]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 6 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_05.75.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 7 : Registered Grid File 3D_06.00.grd as an Output resource with geometry Surface [n_objects = 8]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 7 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_06.00.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 8 : Registered Grid File 3D_06.25.grd as an Output resource with geometry Surface [n_objects = 9]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 8 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_06.25.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 9 : Registered Grid File 3D_06.50.grd as an Output resource with geometry Surface [n_objects = 10]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 9 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_06.50.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 10 : Registered Grid File 3D_06.75.grd as an Output resource with geometry Surface [n_objects = 11]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 10 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_06.75.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 11 : Registered Grid File 3D_07.00.grd as an Output resource with geometry Surface [n_objects = 12]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 11 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_07.00.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 12 : Registered Grid File 3D_07.25.grd as an Output resource with geometry Surface [n_objects = 13]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 12 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_07.25.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 13 : Registered Grid File 3D_07.50.grd as an Output resource with geometry Surface [n_objects = 14]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 13 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_07.50.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 14 : Registered Grid File 3D_07.75.grd as an Output resource with geometry Surface [n_objects = 15]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 14 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_07.75.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 15 : Registered Grid File 3D_08.00.grd as an Output resource with geometry Surface [n_objects = 16]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 15 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_08.00.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 16 : Registered Grid File 3D_08.25.grd as an Output resource with geometry Surface [n_objects = 17]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 16 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_08.25.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 17 : Registered Grid File 3D_08.50.grd as an Output resource with geometry Surface [n_objects = 18]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 17 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_08.50.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 18 : Registered Grid File 3D_08.75.grd as an Output resource with geometry Surface [n_objects = 19]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 18 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_08.75.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 19 : Registered Grid File 3D_09.00.grd as an Output resource with geometry Surface [n_objects = 20]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 19 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_09.00.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 20 : Registered Grid File 3D_09.25.grd as an Output resource with geometry Surface [n_objects = 21]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 20 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_09.25.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 21 : Registered Grid File 3D_09.50.grd as an Output resource with geometry Surface [n_objects = 22]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 21 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_09.50.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 22 : Registered Grid File 3D_09.75.grd as an Output resource with geometry Surface [n_objects = 23]
Errors while running CTest
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 22 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_09.75.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: Object ID 23 : Registered Grid File 3D_10.00.grd as an Output resource with geometry Surface [n_objects = 24]
greenspline [DEBUG]: gmtapi_export_grid: Passed ID = 23 and mode = 0
greenspline [INFORMATION]: Writing grid to file 3D_10.00.grd
greenspline [DEBUG]: region: 0 0, grid: 12 32
greenspline [DEBUG]: -> region: 12 32, grid: 12 32
greenspline [DEBUG]: row: 0 24, col: 0 80
greenspline [DEBUG]: netCDF Library version: 4
greenspline [DEBUG]: packed z-range: [5.96489,6.74662]
greenspline [DEBUG]: GMT_End_IO: Output resource access is now disabled
greenspline [DEBUG]: GMT_Destroy_Data: freed memory for a Cube for object 1
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 1 [n_objects = 23]
greenspline [DEBUG]: gmtlib_unregister_io: Object no 1 has non-NULL resource pointer
greenspline [DEBUG]: GMT_End_IO: Output resource access is now disabled
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 0 [n_objects = 22]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 2 [n_objects = 21]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 3 [n_objects = 20]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 4 [n_objects = 19]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 5 [n_objects = 18]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 6 [n_objects = 17]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 7 [n_objects = 16]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 8 [n_objects = 15]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 9 [n_objects = 14]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 10 [n_objects = 13]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 11 [n_objects = 12]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 12 [n_objects = 11]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 13 [n_objects = 10]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 14 [n_objects = 9]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 15 [n_objects = 8]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 16 [n_objects = 7]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 17 [n_objects = 6]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 18 [n_objects = 5]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 19 [n_objects = 4]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 20 [n_objects = 3]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 21 [n_objects = 2]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 22 [n_objects = 1]
greenspline [DEBUG]: gmtlib_unregister_io: Unregistering object no 23 [n_objects = 0]
greenspline (gmtlib_free_tmp_arrays): tried to free unallocated memory
gmt [DEBUG]: Exit:  gmt_reload_settings
gmt [DEBUG]: Enter: gmtlib_plot_C_format
gmt [DEBUG]: Exit:  gmtlib_plot_C_format
gmt [DEBUG]: Enter: gmtinit_get_history
gmt [DEBUG]: Enter: gmt_hash_init
gmt [DEBUG]: Exit:  gmt_hash_init
gmt [DEBUG]: Exit:  gmtinit_get_history
gmt [DEBUG]: Initialize FFTW with 2 threads.
gmt [DEBUG]: GMT_Create_Session initialized GMT structure
gmt [DEBUG]: Loading core GMT shared library: libgmt.so
gmt [DEBUG]: Shared Library # 0 (core). Path = libgmt.so
gmt [DEBUG]: Loading GMT plugins from: /home/runner/work/gmt/gmt/build/src/plugins
gmt [DEBUG]: Shared Library # 1 (supplements). Path = /home/runner/work/gmt/gmt/build/src/plugins/supplements.so
gmt [DEBUG]: Revised options: -R12/32/0/6/5/10 -JX6i/3i -JZ2.5i -p200/25 -Bx5f1g1 -By1g1 -Bz2f1 -BWSneZ+b -P -K
gmt (gmtlib_free_tmp_arrays): tried to free unallocated memory
psbasemap [DEBUG]: History: Process -R12/32/0/6/5/10
psbasemap [DEBUG]: History: Process -JX6i/3i
psbasemap [DEBUG]: History: Process -JZ2.5i
psbasemap [DEBUG]: History: Process -p200/25
psbasemap [DEBUG]: History: Process -Bx5f1g1
psbasemap [DEBUG]: History: Process -By1g1
psbasemap [DEBUG]: History: Process -Bz2f1
psbasemap [DEBUG]: History: Process -BWSneZ+b
psbasemap [DEBUG]: gmtparse_B_arg_inspector: GMT4 = 0 vs. GMT = 1
psbasemap [DEBUG]: gmtparse_B_arg_inspector: Detected GMT >=5 style elements in -B option
psbasemap [DEBUG]: gmtparse_B_arg_inspector: GMT4 = 0 vs. GMT = 1
psbasemap [DEBUG]: gmtparse_B_arg_inspector: Detected GMT >=5 style elements in -B option
psbasemap [DEBUG]: gmtparse_B_arg_inspector: GMT4 = 0 vs. GMT = 1
psbasemap [DEBUG]: gmtparse_B_arg_inspector: Detected GMT >=5 style elements in -B option
psbasemap [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt
psbasemap [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/work/gmt/gmt/test/greenspline
psbasemap [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt/cache
psbasemap [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt/server
psbasemap [DEBUG]: Got regular w/e/s/n for region (12/32/0/6/5/10)
psbasemap [INFORMATION]: Constructing the basemap
psbasemap [DEBUG]: Projected values in meters: 12 32 0 6
psbasemap [INFORMATION]: Linear projection implies y-axis distance exaggeration relative to the x-axis by a factor of 1.66667
psbasemap [INFORMATION]: Map scale is 0.00333333 km per inch or 1:131.234.
psbasemap [DEBUG]: Running in PS mode classic
psbasemap [DEBUG]: Basemap order: Frame = below  Grid = below  Tick/ANot = below
psbasemap [DEBUG]: Entering gmtplot_map_gridlines
psbasemap [INFORMATION]: Save current gridline information to gmt.history
psbasemap [DEBUG]: Draw x = 12 from 0 to 6
psbasemap [DEBUG]: Draw x = 13 from 0 to 6
psbasemap [DEBUG]: Draw x = 14 from 0 to 6
psbasemap [DEBUG]: Draw x = 15 from 0 to 6
psbasemap [DEBUG]: Draw x = 16 from 0 to 6
psbasemap [DEBUG]: Draw x = 17 from 0 to 6
psbasemap [DEBUG]: Draw x = 18 from 0 to 6
psbasemap [DEBUG]: Draw x = 19 from 0 to 6
psbasemap [DEBUG]: Draw x = 20 from 0 to 6
psbasemap [DEBUG]: Draw x = 21 from 0 to 6
psbasemap [DEBUG]: Draw x = 22 from 0 to 6
psbasemap [DEBUG]: Draw x = 23 from 0 to 6
psbasemap [DEBUG]: Draw x = 24 from 0 to 6
psbasemap [DEBUG]: Draw x = 25 from 0 to 6
psbasemap [DEBUG]: Draw x = 26 from 0 to 6
psbasemap [DEBUG]: Draw x = 27 from 0 to 6
psbasemap [DEBUG]: Draw x = 28 from 0 to 6
psbasemap [DEBUG]: Draw x = 29 from 0 to 6
psbasemap [DEBUG]: Draw x = 30 from 0 to 6
psbasemap [DEBUG]: Draw x = 31 from 0 to 6
psbasemap [DEBUG]: Draw x = 32 from 0 to 6
psbasemap [DEBUG]: Draw y = 0 from 12 to 32
psbasemap [DEBUG]: Draw y = 1 from 12 to 32
psbasemap [DEBUG]: Draw y = 2 from 12 to 32
psbasemap [DEBUG]: Draw y = 3 from 12 to 32
psbasemap [DEBUG]: Draw y = 4 from 12 to 32
psbasemap [DEBUG]: Draw y = 5 from 12 to 32
psbasemap [DEBUG]: Draw y = 6 from 12 to 32
psbasemap [DEBUG]: Exiting gmtplot_map_gridlines
psbasemap (gmtplot_z_gridlines): tried to free unallocated memory
psbasemap (gmtplot_z_gridlines): tried to free unallocated memory
psbasemap (gmtplot_z_gridlines): tried to free unallocated memory
psbasemap (gmtplot_z_gridlines): tried to free unallocated memory
psbasemap (Free_Ctrl): tried to free unallocated memory
psbasemap (Free_Ctrl): tried to free unallocated memory
psbasemap (Free_Ctrl): tried to free unallocated memory
psbasemap (gmtlib_free_tmp_arrays): tried to free unallocated memory
gmt [DEBUG]: Exit:  gmt_reload_settings
gmt [DEBUG]: Enter: gmtlib_plot_C_format
gmt [DEBUG]: Exit:  gmtlib_plot_C_format
gmt [DEBUG]: Enter: gmtinit_get_history
gmt [DEBUG]: Enter: gmt_hash_init
gmt [DEBUG]: Exit:  gmt_hash_init
gmt [DEBUG]: Exit:  gmtinit_get_history
gmt [DEBUG]: Initialize FFTW with 2 threads.
gmt [DEBUG]: GMT_Create_Session initialized GMT structure
gmt [DEBUG]: Loading core GMT shared library: libgmt.so
gmt [DEBUG]: Shared Library # 0 (core). Path = libgmt.so
gmt [DEBUG]: Loading GMT plugins from: /home/runner/work/gmt/gmt/build/src/plugins
gmt [DEBUG]: Shared Library # 1 (supplements). Path = /home/runner/work/gmt/gmt/build/src/plugins/supplements.so
gmt [DEBUG]: Local file /home/runner/.gmt/server/gmt_data_server.txt found
gmt [DEBUG]: File /home/runner/.gmt/server/gmt_data_server.txt less than 24 hours old, refresh is premature.
gmt [DEBUG]: Load contents from /home/runner/.gmt/server/gmt_data_server.txt
gmt [DEBUG]: Local file /home/runner/.gmt/server/gmt_hash_server.txt found
gmt [DEBUG]: File /home/runner/.gmt/server/gmt_hash_server.txt less than 24 hours old, refresh is premature.
gmt [DEBUG]: Revised options: -R -JX -JZ -p200/25 -O -K @Table_5_23.txt -Su0.05i -Gblack -Wfaint
gmt (gmtlib_free_tmp_arrays): tried to free unallocated memory
psxyz [DEBUG]: History: Process -R
psxyz [DEBUG]: History: Process -JX
psxyz [DEBUG]: History: Process -JZ
psxyz [DEBUG]: History: Process -p200/25
psxyz [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt
psxyz [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/work/gmt/gmt/test/greenspline
psxyz [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt/cache
psxyz [DEBUG]: Look for file 12/32/0/6/5/10 in /home/runner/.gmt/server
psxyz [DEBUG]: Got regular w/e/s/n for region (12/32/0/6/5/10)
psxyz [DEBUG]: Remote file @Table_5_23.txt exists locally as /home/runner/.gmt/cache/Table_5_23.txt
psxyz [DEBUG]: Replace file @Table_5_23.txt with /home/runner/.gmt/cache/Table_5_23.txt
psxyz [INFORMATION]: Processing input table data
psxyz [DEBUG]: Projected values in meters: 12 32 0 6
psxyz [INFORMATION]: Linear projection implies y-axis distance exaggeration relative to the x-axis by a factor of 1.66667
psxyz [INFORMATION]: Map scale is 0.00333333 km per inch or 1:131.234.
psxyz [DEBUG]: 3-D shading illusion: lux[k] = -0.13603
psxyz [DEBUG]: 3-D shading illusion: lux[k] = 0.5
psxyz [DEBUG]: 3-D shading illusion: lux[k] = -0.00376576
psxyz [DEBUG]: Running in PS mode classic
psxyz [DEBUG]: Basemap order: Frame = below  Grid = below  Tick/ANot = below
psxyz [DEBUG]: gmtapi_init_import: Passed family = Data Table and geometry = Point
psxyz [DEBUG]: Given full path to file /home/runner/.gmt/cache/Table_5_23.txt
psxyz [DEBUG]: Found readable file /home/runner/.gmt/cache/Table_5_23.txt
psxyz [DEBUG]: Object ID 0 : Registered Data Table File /home/runner/.gmt/cache/Table_5_23.txt as an Input resource with geometry Point [n_objects = 1]
psxyz (gmtapi_init_import): tried to free unallocated memory
psxyz [DEBUG]: gmtapi_init_import: Added 1 new sources
psxyz [DEBUG]: GMT_Init_IO: Returned first Input object ID = 0
psxyz [DEBUG]: GMT_Begin_IO: Mode value 1 not considered (ignored)
psxyz [DEBUG]: GMT_Begin_IO: Initialize record-by-record access for Input
psxyz [DEBUG]: gmtapi_next_io_source: Selected object 0
psxyz [DEBUG]: Given full path to file /home/runner/.gmt/cache/Table_5_23.txt
psxyz [DEBUG]: Found readable file /home/runner/.gmt/cache/Table_5_23.txt
psxyz [DEBUG]: Found readable file /home/runner/.gmt/cache/Table_5_23.txt
psxyz [INFORMATION]: Reading Data Table from file /home/runner/.gmt/cache/Table_5_23.txt
psxyz [DEBUG]: GMT_Begin_IO: Input resource access is now enabled [record-by-record]
psxyz [DEBUG]: Source col types: (Number,Number,Number,Number)
psxyz [DEBUG]: ASCII source scanned: Numerical columns: 4, Trailing text: N, Record type: Numerical only
psxyz [DEBUG]: dist[0] = -0.572452 dist[1] = 0.211309
psxyz [DEBUG]: dist[0] = -1.61475 dist[1] = 0.739582
psxyz [DEBUG]: dist[0] = -1.71736 dist[1] = 0.211309
psxyz [DEBUG]: dist[0] = -2.52473 dist[1] = 0.845237
psxyz [DEBUG]: dist[0] = -1.45272 dist[1] = 0.528273
psxyz [DEBUG]: dist[0] = -2.23039 dist[1] = 0.422618
psxyz [DEBUG]: dist[0] = -2.23039 dist[1] = 1.05655
psxyz [DEBUG]: dist[0] = -3.37529 dist[1] = 0.633927
psxyz [DEBUG]: dist[0] = -2.74342 dist[1] = 0.316964
psxyz [DEBUG]: dist[0] = -2.6111 dist[1] = 0.845237
psxyz [DEBUG]: dist[0] = -2.24386 dist[1] = 0.316964
psxyz [DEBUG]: dist[0] = -2.34647 dist[1] = 0.422618
psxyz [DEBUG]: dist[0] = -3.15384 dist[1] = 0.845237
psxyz [DEBUG]: dist[0] = -4.06382 dist[1] = 0.422618
psxyz [DEBUG]: GMT_End_IO: Input resource access is now disabled
psxyz [DEBUG]: gmtlib_unregister_io: Unregistering object no 0 [n_objects = 0]
psxyz (gmtlib_free_tmp_arrays): tried to free unallocated memory
gmt [DEBUG]: Exit:  gmt_reload_settings
gmt [DEBUG]: Enter: gmtlib_plot_C_format
gmt [DEBUG]: Exit:  gmtlib_plot_C_format
gmt [DEBUG]: Enter: gmtinit_get_history
gmt [DEBUG]: Enter: gmt_hash_init
gmt [DEBUG]: Exit:  gmt_hash_init
gmt [DEBUG]: Exit:  gmtinit_get_history
gmt [DEBUG]: Initialize FFTW with 2 threads.
gmt [DEBUG]: GMT_Create_Session initialized GMT structure
gmt [DEBUG]: Loading core GMT shared library: libgmt.so
gmt [DEBUG]: Shared Library # 0 (core). Path = libgmt.so
gmt [DEBUG]: Loading GMT plugins from: /home/runner/work/gmt/gmt/build/src/plugins
gmt [DEBUG]: Shared Library # 1 (supplements). Path = /home/runner/work/gmt/gmt/build/src/plugins/supplements.so
gmt [DEBUG]: Revised options: -JX 3D_05.00.grd -C10 -L9/11 -S8 -Ddump
gmt (gmtlib_free_tmp_arrays): tried to free unallocated memory
grdcontour [DEBUG]: History: Process -JX
grdcontour [DEBUG]: Found readable file 3D_05.00.grd
grdcontour [DEBUG]: Replace file 3D_05.00.grd with path 3D_05.00.grd
grdcontour [DEBUG]: Replace file 3D_05.00.grd with 3D_05.00.grd
grdcontour [DEBUG]: Look for file 10 in /home/runner/.gmt
grdcontour [DEBUG]: Look for file 10 in /home/runner/work/gmt/gmt/test/greenspline
grdcontour [DEBUG]: Look for file 10 in /home/runner/.gmt/cache
grdcontour [DEBUG]: Look for file 10 in /home/runner/.gmt/server
grdcontour [INFORMATION]: Processing input grid
grdcontour [INFORMATION]: With -D, no plotting will take place
grdcontour [DEBUG]: Found readable file 3D_05.00.grd
grdcontour [DEBUG]: Replace file 3D_05.00.grd with path 3D_05.00.grd
grdcontour [DEBUG]: Found readable file 3D_05.00.grd
grdcontour [DEBUG]: Replace file 3D_05.00.grd with path 3D_05.00.grd
grdcontour [DEBUG]: Found readable file 3D_05.00.grd
grdcontour [DEBUG]: Object ID 0 : Registered Grid File 3D_05.00.grd as an Input resource with geometry Surface [n_objects = 1]
grdcontour [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdcontour [DEBUG]: gmtapi_import_grid: Passed ID = 0 and mode = 1
grdcontour [DEBUG]: Found readable file 3D_05.00.grd
grdcontour [INFORMATION]: Cartesian input grid
grdcontour [DEBUG]: GMT_End_IO: Input resource access is now disabled
grdcontour [DEBUG]: gmtapi_begin_io: Input resource access is now enabled [container]
grdcontour [DEBUG]: gmtapi_import_grid: Passed ID = 0 and mode = 2
grdcontour [INFORMATION]: Reading grid from file 3D_05.00.grd
grdcontour [DEBUG]: packed z-range: [5.96489,6.74662]
grdcontour [INFORMATION]: Cartesian input grid
grdcontour [DEBUG]: Chosen boundary condition for all edges: natural
grdcontour [INFORMATION]: Set boundary condition for all edges: natural
grdcontour [INFORMATION]: Set boundary condition for left   edge: natural
grdcontour [INFORMATION]: Set boundary condition for right  edge: natural
grdcontour [INFORMATION]: Set boundary condition for bottom edge: natural
grdcontour [INFORMATION]: Set boundary condition for top    edge: natural
grdcontour [DEBUG]: GMT_End_IO: Input resource access is now disabled
Warning: r [WARNING]: No contours within specified -L range
grdcontour [DEBUG]: GMT_Destroy_Data: freed memory for a Grid for object 0
grdcontour [DEBUG]: gmtlib_unregister_io: Unregistering object no 0 [n_objects = 0]
grdcontour [DEBUG]: gmtlib_unregister_io: Object no 0 has non-NULL resource pointer
grdcontour (gmtlib_free_tmp_arrays): tried to free unallocated memory
gawk: fatal: cannot open file `dump' for reading (No such file or directory)
ERROR: gspline_4.sh:28
memtrack errors: 0
exit status: 1


0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.67 sec

The following tests FAILED:
	517 - test/greenspline/gspline_4.sh (Failed)
Error: Process completed with exit code 8.
```

--- a/test/greenspline/gspline_4.sh
+++ b/test/greenspline/gspline_4.sh
@@ -24,8 +24,10 @@ z=5
 while read grid; do
 #	echo "Doing z = $z"
 	gmt grdcontour -JX $grid -C10 -L9/11 -S8 -Ddump
-	$AWK '{if ($1 == ">") {print $0} else {print $1, $2, '$z'}}' dump > tmp
-	gmt psxyz -R$R2D/$Z -JX -JZ -p$view -O -K tmp -Gp39+r300+fgray+b- -Wthin >> $ps
+	if [ -s dump ]; then
+		$AWK '{if ($1 == ">") {print $0} else {print $1, $2, '$z'}}' dump > tmp
+		gmt psxyz -R$R2D/$Z -JX -JZ -p$view -O -K tmp -Gp39+r300+fgray+b- -Wthin >> $ps
+	fi
 	z=$(gmt math -Q $z $dz ADD =)
 done < t.lis
 echo "12 6 Volume exceeding 10% UO@-2@- concentration" | gmt pstext -R$R2D/$Z -JX -JZ -p$view -F+jLT+f16p -O -K -Z10 -Dj0.1i >> $ps

